server/services/memory/migration_task.py

```
@@ -0,0 +1,157 @@
+
+logger = logging.getLogger(__name__)
+
+
+@celery.task(name="migrate_kb_to_memory", bind=True, max_retries=0)
+def migrate_kb_to_memory(self):
+    """Migrate all existing KB data into memory artifacts."""
+    stats = {"context": 0, "infrastructure": 0, "documents": 0, "errors": 0}
+
+    try:
+        with db_pool.get_admin_connection() as conn:
+            with conn.cursor() as cursor:
```

Ruff (0.15.18)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)